sympy/matrices/reductions.py

```diff
@@ -299,11 +323,18 @@ def _rref(M, iszerofunc=_iszero, simplify=False, pivots=True,
     if you depend on the form row reduction algorithm leaves entries
     of the matrix, set ``noramlize_last=False``
     """
-
-    simpfunc = simplify if isinstance(simplify, FunctionType) else _simplify
-
-    mat, pivot_cols, _ = _row_reduce(M, iszerofunc, simpfunc,
-            normalize_last, normalize=True, zero_above=True)
+    flag = True
+    if M.is_square and _check_expr(M):
+        DOM = DomainMatrix.from_Matrix(M, radicals_limit=1, field=True)
```

Comment:
Why does this check if `M` is square?

Reply:
In order to avoid non-rectangular matrices.
Kindly refer https://github.com/sympy/sympy/pull/21056#issuecomment-798375335
Thanks.

Reply:
It seems I got confused between this and the charpoly PR. We can compute RREF with non-square matrices.

Reply:
On removing we face an error: 
If we try to construct `DomainMatrix` of the matrix `Matrix([[1, -1, 8], [-sqrt(3), tan(13*pi/45), 0]])`, 
it rasies `NotAlgebraic: tan(13*pi/45) doesn't seem to be an algebraic element`.

Reply:
That could be a polys/DomainMatrix bug. How would you reproduce that with DomainMatrix on master?

Reply:
The main issue is if we consider `extension=True` in `from_Matrix`, it raises the error. 
Since the matrix contains `sqrt(3)`, the radicals_limit being greater than len(radicals), it sets `extension=True`.
If we construct without `extension=True`, it forms the `DomainMatrix` in the `EX` domain and works perfectly.

Reply:
I have opened a seperate issue for this as #21430.
Thanks.

Reply:
The code to count radicals should really count irrational algebraic expressions I guess. It probably doesn't count `tan(13*pi/45)`.